### PR TITLE
[Feature][seatunnel-transforms] Add Uuid transform for spark

### PR DESCRIPTION
**Pull request by wuzhenhua01**
Add Uuid transform for spark and fix some error at replace doc https://github.com/apache/incubator-seatunnel/issues/1762 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-uuid/src/main/scala/org/apache/seatunnel/spark/transform/Uuid.scala [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import org.apache.seatunnel.spark.{BaseSparkTransform, SparkEnvironment}
+import org.apache.spark.sql.{Dataset, Row}
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.functions.{col, udf}
+
+class Uuid extends BaseSparkTransform {
+  private var prng: RandomGenerator = _
+
+  override def process(df: Dataset[Row], env: SparkEnvironment): Dataset[Row] = {
+    val key = config.getString("fields")
```
Comment (ruanwenjun) [inline seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-uuid/src/main/scala/org/apache/seatunnel/spark/transform/Uuid.scala]: Please split this constant string, we don't want to receive another PR to solve this. And split the constant string in `Replace`.

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-uuid/src/main/scala/org/apache/seatunnel/spark/transform/Uuid.scala [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  def generate(prefix: String): String = {
+    val uuid = if (prng == null) UUID.randomUUID else new UUID(prng.nextLong, prng.nextLong)
+    prefix + uuid
+  }
+
+  @VisibleForTesting
+  def setPrng(prng: RandomGenerator): Unit = {
+    this.prng = prng
+  }
+}
```
Comment (ruanwenjun) [inline seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-uuid/src/main/scala/org/apache/seatunnel/spark/transform/Uuid.scala]: Please implement the `getPluginName()` method. And please solve this in `Replace` together.

File: docs/en/transform/uuid.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+The nonSecure algorithm uses a secure random seed but is otherwise deterministic
+
+### common options [string]
+
+Transform plugin common parameters, please refer to [Transform Plugin](common-options.mdx) for details
+
+## Examples
+
+```bash
+  uuid {
```
Comment (ruanwenjun) [inline docs/en/transform/uuid.md]: Maybe use `UUID` is better? Anyway there is a typo.
```suggestion
  UUID {
```

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-uuid/src/main/scala/org/apache/seatunnel/spark/transform/Uuid.scala (hunk 1 above)
Comment (wuzhenhua01) [reply seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-uuid/src/main/scala/org/apache/seatunnel/spark/transform/Uuid.scala]: I'm afraid I don't quite follow your gist. Your meaning we need a list to accept fields ？

Comment (ruanwenjun) [reply seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-uuid/src/main/scala/org/apache/seatunnel/spark/transform/Uuid.scala]: @wuzhenhua01 Sorry, I don't make myself clear. It's better to split the magic string from this method.  We can add an Object e.g. Config or Constant to store these configs key "fields", "prefix".

Comment (wuzhenhua01) [reply seatunnel-transforms/seatunnel-transforms-spark/seatunnel-transform-spark-uuid/src/main/scala/org/apache/seatunnel/spark/transform/Uuid.scala]: Got it, I will refactor later.

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/constants/TransformConfigConstants.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    /* split transform constant */
+    public static final String SPLIT_SEPARATOR = "separator";
+    public static final String DEFAULT_SPLIT_SEPARATOR = " ";
+
+    /* UUID transform constant */
+    public static final String UUID_PREFIX = "prefix";
+    public static final String DEFAULT_UUID_PREFIX = "";
+    public static final String UUID_SECURE = "secure";
+    public static final boolean DEFAULT_UUID_SECURE = false;
```
Comment (ruanwenjun) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/constants/TransformConfigConstants.java]: The constants file need to put into plugin module, you need to create constant file for each plugin. If we put all constant into one file, the file will be very big.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**